### DISCUSSION TITLE
BinarySwitchCCAPI set returns either SupervisionResult or undefined when succeeded

### DISCUSSION THREAD

**Discussion by geonoemi** [zwave-js/zwave-js#6515] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
We use the BinarySwitchCCAPI.set() to turn on/off zwave devices with the following conditions and return values:

- Aeotec smart switch (working device) return value: undefined
- Fibaro smart switch (working device) return value: undefined
- NEO smart switch (working device) return value: { status: 255 } 

When the devices work, and the set function is executed, all devices are turned off/on by the set function, so it works as intended, but it returns undefined for the Aeotec smart switch, while it returns { status: 255 } for the NEO smart switch. The weird thing is that it returns undefined for the working Fibaro smart switch and the working Aeotec one as well.

What did you expect to happen instead?
We would like the set function to return values consequently, so if it succeeded, it should not return undefined.

### Software versions

Driver (node-zwave-js): ZMEEUZBB

### Device information

Manufacturer: Aeotec
Model name: Smart switch 6 ZW096

Manufacturer: Fibaro
Model name: Wall plug FGWP102

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

[a comment by an automated account is not shown]

**Comment by AlCalzone** [comment, marked as answer]
This is intended in order to be able to distinguish between unsupervised and supervised commands. You can use the `isUnsupervisedOrSucceeded` method from `@zwave-js/core` to check if a command was successful, supervised or not.

**Reply by geonoemi** [reply]
Many thanks!
